packages/core/src/objects/_abap_object.ts

```diff
@@ -43,6 +46,28 @@ export abstract class ABAPObject extends AbstractObject {
     return {updated: true, runtime: result.runtime};
   }
 
+  private withoutGloballyExcluded(globalExclude: readonly string[], files: IFile[]): IFile[] {
+
+    if (!globalExclude || globalExclude.length === 0) {
+      return files;
+    }
+
+    const globalExcludePatterns = (globalExclude).map(pattern => new RegExp(pattern, "i"));
+    const afterExclude: IFile[] = [];
+
+    for (const file of files) {
+      for (const exclude of globalExcludePatterns) {
+        if (exclude.exec(file.getFilename())) {
+          continue;
+        } else {
+          afterExclude.push(file);
```

Comment:
if there are multiple globalExcludePatterns, then the files will be pushed multiple times?

Reply:
fixed, also cleaned up some of the logic